### ISSUE TITLE
Node.js: Optional array documentation designation is wrong.

### ISSUE THREAD

**Issue by lukesneeringer** [googleapis/toolkit#1638]
The JSDoc build on logging is barfing with [an error](https://circleci.com/gh/googleapis/nodejs-logging/205).

While attempting to fix said error, I gallantly (cue awesome "the cavalry is coming" music here) tracked down the problem to [this spot](https://github.com/googleapis/toolkit/blob/master/src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSApiMethodParamTransformer.java#L82-L87). The problem is that I have no clue how to actually fix it. :thinking: 

Basically: We should not append the `=` if it is an array type, because `string[]=` (or, for that matter, `${anything}[]=`) is not valid.

Please do the needful.

**Comment by jmdobry** [comment]
We should just not use `=` to denote optional arguments at all, instead we should wrap the full argument name in square brackets, e.g.

```
* @param {string} requiredArg
* @param {string[]=} optionalArg
* @param {object=} options
* @param {boolean=} options.autoCreate
```
becomes
```
* @param {string} requiredArg
* @param {string[]} [optionalArg]
* @param {object} [options]
* @param {boolean} [options.autoCreate]
```
